scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py

```diff
@@ -0,0 +1,233 @@
+        """
+        selection = {}
+        for index, fit in enumerate(self._fit_context.fit_list):
+            if fit.fit_function_name != self.selected_fit_function():
+                continue
+            name = fit.input_workspace
+            if name in existing_selection:
+                checked = existing_selection[name][1]
+            else:
+                checked = True
+            selection[name] = [index, checked, True]
+
+        return selection
+
+    def log_selection(self, existing_selection):
+        """
+        Combine the existing selection state of log values with the the set for the current
+        workspace at the top of the list
+        :param existing_selection: A dict defining any current selection model. The
+        format matches that of the ListSelectorPresenter class' model.
+        :return: The logs in the first workspace along with their selection status. The
+        format matches that of the ListSelectorPresenter class' model.
+        """
+        selection = {}
+        fits = self._fit_context.fit_list
+        if not fits:
+            return selection
+
+        logs = log_names(fits[0].input_workspace)
+        for index, name in enumerate(logs):
+            if name in existing_selection:
+                checked = existing_selection[name][1]
+            else:
+                checked = False
+            selection[name] = [index, checked, True]
+
+        return selection
+
+    # Ideally log_selection and model_selection would be part of this model but the ListSelectorPresenter
+    # contains the model and it's not yet trivial to share that
+    def create_results_table(self, log_selection, results_selection):
+        """Create a TableWorkspace with the fit results and logs combined. The format is
+        a single row per workspace with columns:
+          |workspace_name|selected_log_1|selected_log_2|...|param1_|param1_err|param2|param2_err|...|
+        Any time-series log values are averaged.
+        The workspace is added to the ADS with the name given by results_table_name
+
+        :param log_selection: The current selection of logs as a list
+        It is assumed this is ordered as it should be displayed. It can be empty.
+        :param results_selection: The current selection of result workspaces as a list of 2-tuple
+        [(workspace, fit_position),...]
+        It is assumed this is not empty and ordered as it should be displayed.
+        """
+        results_table = self._create_empty_results_table(
+            log_selection, results_selection)
+
+        ads = AnalysisDataService.Instance()
+        fit_list = self._fit_context.fit_list
+        for _, position in results_selection:
+            fit = fit_list[position]
+            row_dict = {'workspace_name': fit.parameter_name}
+            # logs first
+            if len(log_selection) > 0:
+                workspace = ads[fit.input_workspace]
```

Comment:
I believe we can grab this from the MuonWorkspaceWrapper, a job for later.

Reply:
Ah I see. I hadn't appreciated that. :+1: